### PR TITLE
Image processing and file size validation

### PR DESCRIPTION
**Pull request by C-Anirudh**
Used `django-imagekit` to compress the uploaded images, without adversely affecting their quality and converted each uploaded image to JPEG format (best format for lossy compression).

Also created `validators.py` file in `gallery` app which limits the size of any uploaded image or file to 2MB (can be modified as per requirement later), and displays an error message if the size limit is exceeded.
Sample error message :
file_error

### CODE REVIEW

**Comment by aswinshenoy** [comment]
This pull request **fixes 5 alerts** when merging 73a35f2291fce2cd0dcd4377d66b968b50457622 into 33efb44e13f989b81a4427341f457c1f2ca3c0de - [view on LGTM.com](https://lgtm.com/projects/g/amfoss/cms/rev/pr-8cbc4b85ada085b1b6805c569920fe48bd656e0d)

**fixed alerts:**

* 5 for Unused import

---

*Comment posted by [LGTM.com](https://lgtm.com)*

File: activity/models.py [hunk 1]
```diff
@@ -5,8 +5,14 @@
 from members.models import Skill, Portal, Organization
 from datetime import date
 import uuid
+from gallery.validators import validate_file_size
+from imagekit.models import ProcessedImageField
 
 EVENT_TYPES = (('C', 'Conference/Summit'), ('H', 'Hackathon'), ('D', 'Developer Meet'), ('I', 'Internship'), ('E','Student Exchange'), ('S','Summer School'), ('O', 'Others'))
+processed_image_field_specs = {
```
Comment (aswinshenoy) [inline activity/models.py]: can't we import this too everywhere?

File: gallery/validators.py [hunk 2]
```diff
@@ -0,0 +1,9 @@
+from django.core.exceptions import ValidationError
+
+def validate_file_size(value):
+    filesize = value.size
+
+    if filesize > 2097152:
+        raise ValidationError("The maximum file size that can be uploaded is 2MB.")
```
Comment (aswinshenoy) [inline gallery/validators.py]: make it 10MB as limit, but it should compress the image well below 500kb.
For documents, keep 10mb as limit, i dont see compression required for documents.

File: status/admin.py [hunk 3]
```diff
@@ -14,7 +14,7 @@ class StatusAdmin(admin.ModelAdmin):
 
 @admin.register(Task)
 class TaskAdmin(admin.ModelAdmin):
-    fields= (('title','date'),('status','tags'),('assignees','team'),'description','updates')
+    fields= (('title','due_date'),('status','tags'),('assignees','team'),'description','updates')
```
Comment (aswinshenoy) [inline status/admin.py:17]: :+1:

**Review by aswinshenoy — CHANGES_REQUESTED** [review]
Can make the changes to allow uploads till 10mb, and put process_image_field_specs together somewhere and import it? 
Also, can you include the before and after size of image upload (or document) as screenshot or something.

**Comment by aswinshenoy** [comment]
This pull request **fixes 5 alerts** when merging 79018eedabe4270be9ebda6c3780cdeb915457e3 into 7a525d8fbf000601f5fa47a9848f1cd802d43daa - [view on LGTM.com](https://lgtm.com/projects/g/amfoss/cms/rev/pr-e355e04b0ffa65130e7f37ad455776b957b63576)

**fixed alerts:**

* 5 for Unused import

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Review by aswinshenoy — APPROVED** [review]

**Comment by aswinshenoy** [comment]
@C-Anirudh  ```django-imagekit``` needs to be installed for the plugin to be running, I think you forgot to add it to the ```requirements.txt``` file.

**Comment by C-Anirudh** [comment]
I have added it, it was visible to me as a change when I created the PR.

**Comment by aswinshenoy** [comment]
@C-Anirudh can you kindly resolve the conflicts too? Just pull the changes from the master to your branch, it should work :+1:

**Comment by aswinshenoy** [comment]
This pull request **fixes 5 alerts** when merging 2ad3f57ab3d133fb0f8a46cee255da8b557dd4be into b624f5b09dbd0271d783f020d6b06e41ff410176 - [view on LGTM.com](https://lgtm.com/projects/g/amfoss/cms/rev/pr-6f38caa0f3d764a5067db1d5375b6c1fe298fef2)

**fixed alerts:**

* 5 for Unused import

---

*Comment posted by [LGTM.com](https://lgtm.com)*

File: blog/models.py [hunk 4]
```diff
@@ -48,9 +51,9 @@ def get_featured_image_path(self, filename):
     author = models.ForeignKey(User, on_delete=models.CASCADE, related_name='Post', verbose_name='Author')
     content = RichTextField()
     date = models.DateField(default=date.today)
-    featured_image = models.ImageField(default='', verbose_name='Featured Image', upload_to=get_featured_image_path)
+    featured_image = ProcessedImageField(default='', verbose_name='Featured Image', upload_to=get_featured_image_path, validators=[validate_file_size], **processed_image_field_specs)
     category = models.ForeignKey(Category, on_delete=models.SET_NULL, verbose_name='Category', null=True, blank=True)
-    tags = models.ManyToManyField(Tag, verbose_name='Tag', blank=True)
+    tags = models.ManyToManyField(Tag,verbose_name='Tag', blank=True)
```
Comment (aswinshenoy) [inline blog/models.py:56]: its always adviced to reformat your code (all the files) before committing, this ensures proper spacing and indentation are in place automatically.

**Review by aswinshenoy — APPROVED** [review]

**Pull request merged**